mindsdb/integrations/handlers/azure_blob_handler/azure_blob_handler.py

```diff
@@ -0,0 +1,364 @@
+        )
+        self.connection = blob_service_client
+        self.is_connected = True
+        return blob_service_client
+
+    def check_connection(self) -> StatusResponse:
+        """ Check connection to the handler
+        Returns:
+            HandlerStatusResponse
+        """
+        response = StatusResponse(False)
+
+        try:
+            client = self.connect()
+            client.get_account_information()
+            response.success = True
+
+        except Exception as e:
+            logger.error(f'Error connecting to Azure Blob: {e}!')
+            response.error_message = e
+
+        self.is_connected = response.success
+        return response
+
+    def native_query(self, query: str = None) -> Response:
+        """Receive raw query and act upon it somehow.
+        Args:
+            query (Any): query in native format (str for sql databases,
+                dict for mongo, api's json etc)
+        Returns:
+            HandlerResponse
+        """
+
+    def call_application_api(self, method_name:str = None, params:dict = None) -> pd.DataFrame:
+        """Receive query as AST (abstract syntax tree) and act upon it somehow.
+        Args:
+            query (ASTNode): sql query represented as AST. Can be any kind
+                of query: SELECT, INSERT, DELETE, etc
+        Returns:
+            DataFrame
+        """
+
+    def disconnect(self):
+        """
+        Closes the connection to the Azure Blob account if it's currently open.
+        """
+        if not self.is_connected:
+            return
+        self.connection.close()
+        self.is_connected = False
+
+    @contextmanager
+    def _connect_duckdb(self):
+        """
+        Creates temporal duckdb database which is able to connect to the Azure Blob account.
+        Have to be used as context manager
+
+        Returns:
+            DuckDBPyConnection
+        """
+        # Connect to Azure Blob via DuckDB.
+        duckdb_conn = duckdb.connect(":memory:")
+        duckdb_conn.execute('INSTALL azure')
+        duckdb_conn.execute('LOAD azure')
+
+        # Configure mandatory credentials.
+        duckdb_conn.execute(f'SET azure_storage_connection_string="{self.connection_string}"')
+
+        try:
+            yield duckdb_conn
+        finally:
+            duckdb_conn.close()
+
+    def read_as_table(self, key) -> pd.DataFrame:
+        """
+        Read object as dataframe. Uses duckdb
+        """
+
+        with self._connect_duckdb() as connection:
+            cursor = connection.execute(f'SELECT * FROM "azure://{self.container_name}/{key}"')
+            return cursor.fetchdf()
+
+    def _read_as_content(self, key) -> None:
+        """
+        Read object as content
+        """
+
+        connection = self.connect()
+        client = connection.get_blob_client(container=self.container_name,blob=key)
+
+        return client.download_blob()
+
+    def add_data_to_table(self, key, df) -> None:
```

Comment:
if this is not needed let's remove it

Reply:
It will be used by `insert` sql statement so maybe I'll implement it here, Zoran <-- I just knew this after read Minura's latest comment